esx_service/vmodl/VsphereContainerServiceImpl.py

```diff
@@ -0,0 +1,269 @@
+# Copyright 2016 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Copyright 2016 VMware, Inc.  All rights reserved. 
+Licensed under the Apache License, Version 2.0 
+http://www.apache.org/licenses/LICENSE-2.0
+"""
+
+import logging
+import os
+import os.path
+import sys
+
+from pyVmomi import Vim, vim, vmodl
+from MoManager import GetMoManager
+
+# Location of utils used by the plugin
+TOP_DIR = "/usr/lib/vmware/vmdkops"
+PY_LOC  = os.path.join(TOP_DIR, "Python")
+PY_BIN  = os.path.join(TOP_DIR, "bin")
+
+# vmdkops python utils are in PY_LOC, so insert to path ahead of other stuff
+sys.path.insert(0, PY_LOC)
+sys.path.insert(0, PY_BIN)
+
+import error_code
+from error_code import ErrorCode
+import auth_api
+import vmdk_utils
+
+TENANT_NAME_MAX_LEN = 64
+TENANT_DESC_MAX_LEN = 256
+
+class TenantManagerImpl(vim.vcs.TenantManager):
+    '''Implementation of VCS TenantManager'''
+
+    def __init__(self, moId):
+        vim.vcs.TenantManager.__init__(self, moId)
+
+    def CreateTenant(self, name, description=None):
+        logging.info("Creating a tenant: name=%s, description=%s", name, description)
+
+        if len(name) == 0:
+            error_info = ErrorCode.VMODL_TENANT_NAME_EMPTY.format(name);
+            return error_info
+
+        if len(name) > TENANT_NAME_MAX_LEN:
+            error_info = ErrorCode.VMODL_TENANT_NAME_TOO_LONG.format(name);
+            return error_info
+
+        if description and len(description) > TENANT_DESC_MAX_LEN:
+            error_info = ErrorCode.VMODL_TENANT_NAME_TOO_LONG.format(name);
+            return error_info
+
+        # Create the tenant in the database
+        error_info, tenant = auth_api._tenant_create(name, description);
+        if error_info:
```

Comment:
Please skip negative scenarios (NotFound, AlreadyExists, InvalidArgument, etc.) for now. I will handle that in a separate PR (or commit).